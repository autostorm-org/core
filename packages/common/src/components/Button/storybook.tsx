import React from "react";
import Button from "./Button";

// This default export determines where you story goes in the story list
export default {
  title: "Button",
  component: Button,
};

export const ButtonStory = () => {
  return <Button>Hello</Button>;
};
