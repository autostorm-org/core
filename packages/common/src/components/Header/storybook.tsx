import React from "react";
import Header from "./Header";
import HeaderLogo from "./HeaderLogo";
import { HeaderLocations } from "./constants";
import HeaderItem from "./HeaderItem";
import Spain from "../../assets/flags/Spain.png";
import Brazil from "../../assets/flags/Brazil.png";
import ResponsiveTemplate from "../ResponsiveTemplate";
import ListFixed from "../ListFixed";
import Icon from "../Icon";
import HeaderItemAbsolute from "./HeaderItemAbsolute";
import { ListFullScreen, ListFullScreenRenderer } from "../List";
import { Menu } from "react-feather";
// This default export determines where you story goes in the story list
export default {
  title: "Header",
  component: Header,
};
declare var require: any;

const logoPath = "./../../assets/autofica-defi-day-logo.svg";

const Desktop = (args) => (
  <Header>
    <HeaderLogo
      logoPath={require("./../../assets/autofica-defi-logo-short-6.svg")}
      itemLocation={HeaderLocations.left}
    />
    <HeaderItem>Loan Manager</HeaderItem>
    <HeaderItem>Whitepaper</HeaderItem>
    <HeaderItem
      dropDown
      sections={[
        {
          links: [
            {
              content: (
                <>
                  <span>br-PT</span>{" "}
                  <Icon
                    src={Brazil}
                    width={23}
                    height={13}
                    alt={"portuguese"}
                  />
                </>
              ),
            },
          ],
        },
      ]}
      Submenu={ListFixed}
    >
      es-ES <Icon src={Spain} width={23} height={13} alt={"espanol"} />
    </HeaderItem>
  </Header>
);

const Mobile = (args) => {
  const [isOpen, setOpen] = React.useState(false);
  return (
    <Header
      submenu={
        isOpen ? (
          <ListFullScreen>
            <ListFullScreenRenderer>Vende</ListFullScreenRenderer>{" "}
            <ListFullScreenRenderer>Contacto</ListFullScreenRenderer>
            <ListFullScreenRenderer>Ayuda</ListFullScreenRenderer>
            <ListFullScreenRenderer>Mi Cuenta</ListFullScreenRenderer>
          </ListFullScreen>
        ) : null
      }
    >
      <HeaderLogo
        logoPath={require("./../../assets/autofica-defi-day-logo.svg")}
        itemLocation={HeaderLocations.center}
        overrides={{ paddingTop: 10, paddingBottom: 10, width: 125 }}
      />
      <HeaderItemAbsolute
        itemLocation={HeaderLocations.right}
        onClick={function () {
          setOpen(!isOpen);
        }}
      >
        <Menu color="white" />
      </HeaderItemAbsolute>
    </Header>
  );
};

const ResponsiveStory = () => (
  <ResponsiveTemplate mobile={Mobile} tablet={Desktop} desktop={Desktop} />
);

export const FirstStory = ResponsiveStory.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
