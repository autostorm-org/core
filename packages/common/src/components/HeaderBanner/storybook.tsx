import React from "react";
import HeaderBannerText from "./HeaderBannerText";
// This default export determines where you story goes in the story list
export default {
  title: "HeaderBannerText",
  component: HeaderBannerText,
};

const Template = (args) => <HeaderBannerText>Hello</HeaderBannerText>;

export const FirstStory = Template.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
