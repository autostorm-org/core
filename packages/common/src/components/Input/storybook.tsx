import React from "react";
import Input from "./Input";
// This default export determines where you story goes in the story list
export default {
  title: "Input",
  component: Input,
};

const Template = () => <Input />;

export const FirstStory = Template.bind({});
