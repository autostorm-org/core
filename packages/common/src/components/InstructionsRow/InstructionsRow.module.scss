@import "../../theme/constants.scss";
.root {
  background-color: var(--instructions-row-background-color);
  padding: 40px 0px 40px 0px;
  @media (min-width: $layout-desktop-min) {
    padding: 12px 0px 12px 0px;
  }
}

.circle {
  width: 60px;
  height: 60px;
  background: var(--instructions-row-circle-background-color) 0% 0% no-repeat
    padding-box;
  border: 3px solid var(--instructions-row-circle-border-color);
  border-radius: 50%;
  opacity: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 10px;
}

.circleText {
  font-weight: var(--instructions-row-circle-font-weight);
}

.instructionsWrapper {
  display: flex;
  justify-content: space-evenly;
  background-color: inherit;
  width: 100%;
  align-items: center;
  flex-flow: row wrap;
  & > :not(:last-child) {
    margin-bottom: 30px;
  }

  @media (min-width: $layout-desktop-min) {
    & > :not(:last-child) {
      margin-bottom: auto;
    }
  }
}

.instructionGroup {
  display: flex;
  align-items: center;
}
.stepText {
  width: 275px;
}
