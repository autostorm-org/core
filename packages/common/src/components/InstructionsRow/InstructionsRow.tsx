import React from "react";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import { BoundedRow } from "../BoundedRow";
import type { WithOverrides } from "../../types";
import styles from "./InstructionsRow.module.scss";

type Step = Readonly<{
  visualOrder: string;
  content: string;
}>;

type InstructionsRowProps = WithOverrides<{
  steps: Array<Step>;
}>;

const InstructionsRow = (props: InstructionsRowProps) => {
  return (
    <div className={`${styles.root} ${props.override || ""}`}>
      <BoundedRow>
        <div className={styles.instructionsWrapper}>
          {props.steps.map((step) => {
            return (
              <div className={styles.instructionGroup}>
                <div className={styles.circle}>
                  <Typography
                    variant={TypographyVariant_ENUM.p}
                    override={styles.circleText}
                  >
                    {step.visualOrder}
                  </Typography>
                </div>
                <Typography
                  variant={TypographyVariant_ENUM.p}
                  override={styles.stepText}
                >
                  {step.content}
                </Typography>
              </div>
            );
          })}
        </div>
      </BoundedRow>
    </div>
  );
};
export default InstructionsRow;
export type { InstructionsRowProps };
