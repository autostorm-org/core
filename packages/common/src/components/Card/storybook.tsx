import React from "react";
import Card from "./Card";
import Typography, { EnumTypographyVariant } from "../Typography";
import Button from "../Button";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { ThemeContext, nightTheme, defaultTheme } from "../../theme";
import ResponsiveTemplate from "../ResponsiveTemplate";
// This default export determines where you story goes in the story list
export default {
  title: "Card",
  component: Card,
  decorators: [withKnobs],
};

const MobileCard = () => {
  return (
    <div style={{ height: 184, width: 350 }}>
      <Card>
        <Typography variant={EnumTypographyVariant.h3}>Card title</Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Card body contents. Card body contents. Card body contents. Card body
          contents.
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 9 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const DesktopCard = () => {
  return (
    <div style={{ height: 240, width: 560 }}>
      <Card style={{ padding: "30px" }}>
        <Typography variant={EnumTypographyVariant.h1}>
          Crypto car loan finder
        </Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Get a collateralized loan for your next car purchase. Chose your
          backing assets. Powered by Multiplier.{" "}
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 10 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const ResponsiveStory = () => {
  const isDayMode = boolean("DAY MODE", true);
  return (
    <ThemeContext.Provider value={isDayMode ? defaultTheme : nightTheme}>
      <ResponsiveTemplate
        mobile={MobileCard}
        desktop={DesktopCard}
        tablet={DesktopCard}
      />
    </ThemeContext.Provider>
  );
};

export const FirstStory = ResponsiveStory.bind({});
