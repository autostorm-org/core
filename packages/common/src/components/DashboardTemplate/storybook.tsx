import React from "react";
import DashboardTemplate from "./DashboardTemplate";
import { BoundedRow } from "../BoundedRow";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import Icon from "../Icon";
import Spain from "../../assets/flags/Spain.png";
import { HeaderResponsive } from "../Header";
import DashboardOptionsColumn from "./DashboardOptionsColumn";
import DashboardOptionsItem from "./DashboardOptionsItem";
import { PillButton } from "./../PillButton";
import { Eye, ArrowLeftCircle, DollarSign, ArrowUp, Key } from "react-feather";

export default {
  title: "Dashboard Template Stories",
};

export const DashboardOptionsColumnStory = () => {
  return (
    <DashboardOptionsColumn>
      <DashboardOptionsItem activeElement href={"#"}>
        hello now
      </DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
    </DashboardOptionsColumn>
  );
};

export const DashboardTemplateStory = () => {
  return (
    <>
      <HeaderResponsive
        options={[
          { content: "Loan Manager", href: "#" },
          { content: "Whitepaper", href: "#" },
          {
            content: (
              <>
                es-ES{" "}
                <Icon src={Spain} width={23} height={13} alt={"espanol"} />
              </>
            ),
          },
        ]}
      />
      <div style={{ marginTop: 30 }}>
        <BoundedRow>
          <DashboardTemplate
            mainContent={
              <Typography
                variant={TypographyVariant_ENUM.h3}
                style={{ padding: 30 }}
              >
                Main Content
              </Typography>
            }
            optionsContent={
              <>
                <DashboardOptionsColumn style={{ borderTopLeftRadius: 8 }}>
                  <div style={{ marginTop: 30, marginBottom: 20 }}>
                    <PillButton
                      onClick={() => {
                        console.log("CLIICK");
                      }}
                    >
                      <Icon
                        src={"crypto-logos/bsc.svg"}
                        height={22}
                        width={22}
                        alt={"BSC logo"}
                        style={{ marginRight: 10 }}
                      />
                      BSC Mainnet
                    </PillButton>
                  </div>

                  <DashboardOptionsItem href={"#"} activeElement>
                    <Eye style={{ marginRight: 10 }} />
                    Loans Overview
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowLeftCircle style={{ marginRight: 10 }} />
                    Borrow
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <DollarSign style={{ marginRight: 10 }} />
                    Payback
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowUp style={{ marginRight: 10 }} />
                    Withdraw Collateral
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <Key style={{ marginRight: 10 }} />
                    Deposit
                  </DashboardOptionsItem>
                </DashboardOptionsColumn>
              </>
            }
          />
        </BoundedRow>
      </div>
    </>
  );
};
