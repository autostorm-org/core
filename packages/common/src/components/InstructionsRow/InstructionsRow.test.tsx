import React from "react";
import renderer from "react-test-renderer";
import { InstructionsRowStory } from "./InstructionsRow.storybook";
describe("Instructions Row Test", () => {
  it("Should render Instructions Row component", () => {
    const root = renderer.create(<InstructionsRowStory />);
    let tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
