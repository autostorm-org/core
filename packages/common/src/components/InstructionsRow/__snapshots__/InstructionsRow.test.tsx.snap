// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions Row Test Should render Instructions Row component 1`] = `
<div
  className="root "
>
  <div
    className="root  "
  >
    <div
      className="instructionsWrapper"
    >
      <div
        className="instructionGroup"
      >
        <div
          className="circle"
        >
          <span
            className="p circleText"
          >
            1
          </span>
        </div>
        <span
          className="p stepText"
        >
          Open the DAPP and connect a BSC wallet. and authorize the DAPP
        </span>
      </div>
      <div
        className="instructionGroup"
      >
        <div
          className="circle"
        >
          <span
            className="p circleText"
          >
            2
          </span>
        </div>
        <span
          className="p stepText"
        >
          Find loans with varying APY and chose the best for your situation.
        </span>
      </div>
      <div
        className="instructionGroup"
      >
        <div
          className="circle"
        >
          <span
            className="p circleText"
          >
            3
          </span>
        </div>
        <span
          className="p stepText"
        >
          Manage your loans and withdraw your collateral once paid
        </span>
      </div>
    </div>
  </div>
</div>
`;
